src/Modules/OrchardCore.Commerce/Services/ShoppingCartHelpers.cs

```diff
@@ -94,6 +95,14 @@ public class ShoppingCartHelpers : IShoppingCartHelpers
             totals = lines.CalculateTotals().ToList();
         }
 
+        for (var i = 0; i < totals.Count; i++)
```

Comment:
Totals is just the sum value. You should update the `LinePrice` and `UnitPrice` in `lines` as well.